### DISCUSSION TITLE
allow sliders with reverse ends to have a shorter-than-normal last reverse

### DISCUSSION THREAD

**Discussion by iSimpFor** [ppy/osu#20147] [Ideas]
in the editor, you can make reverse sliders

you should be able to then shorten the how long the last reverse will last and the slider will shorten to account for how short the last reverse is
 
![shorty](https://user-images.githubusercontent.com/44514667/188464744-37c1186b-25a5-41d3-9ef2-41c89933e634.png)

**Comment by mk56-spn** [comment]
why specifically the last and not let each return be be variable in  length as long as each return is snapped?

**Reply by iSimpFor** [reply]
good idea, may add that after more discussion

if you want that, the reverse can only be as long as it doesn't go past either end of the slider

**Reply by iSimpFor** [reply]
![shorty](https://user-images.githubusercontent.com/44514667/188984743-24b2dd8d-0bcb-4071-b9ce-613f434b7899.png)
i have made a version including the mid-slider reverses
(mid-slider velocity changes will be for another discussion)

**Comment by ghost** [comment]
i feel like this would lead way to some playability issues, and would be hard to read. in particular, im not sure how one would explain this mechanic to new players, like in the tutorial

**Reply by iSimpFor** [reply]
i would think that this mechanic would mostly be used in higher difficulties

for playability, i don't really see any issue since you hold down the slider until it's done regardless of how many slider ends or when they happen and i think it should be up to the mapper to make such sliders readable and playable

i think it can be very good if used right
